dev/cs43l22/export/dev/cs43l22.hpp

```diff
@@ -0,0 +1,838 @@
+        ms1230  = 0x0,  // 1.23 s
+        ms2580  = 0x20, // 2.58 s
+        ms3900  = 0x40, // 3.90 s
+        ms5200  = 0x60, // 5.20 s
+        ms6600  = 0x80, // 6.60 s
+        ms8050  = 0xA0, // 8.05 s
+        ms9350  = 0xC0, // 9.35 s
+        ms10800 = 0xE0, // 10.80 s
+    };
+
+    //! Register masks for operation modes of the beep generator.
+    //!
+    enum class beep_mode
+    {
+        single    = 0x40, // Play one beep
+        multiple  = 0x80, // Play beep for on_time, mute for off_time
+        continuos = 0xC0, // Play continuously
+    };
+
+    //!  Maximum value of the master volume.
+    //! \details The minimum value (0) represents the signal out the DSP: -102 dB.
+    //!  The maximum value (0xE4) represents +12 dB. Step size is 0.5 dB.
+    //!
+    static constexpr uint8_t max_master_volume = 0xE4;
+
+    //!  Maximum value of the headphone volume.
+    //! \details The minimum value (0) represents the signal out the DAC: -127 dB (muted).
+    //!  The maximum value (0xFF) represents the signal out the DAC: 0 dB. Step size is 0.5 dB.
+    //!
+    static constexpr uint8_t max_headphone_volume = 0xFF;
+
+    //!  Maximum value of the beep volume.
+    //! \details The minimum value (0) represents the gain: -56 dB.
+    //!  The maximum value (0x1F) represents the gain: +6 dB. Step size is 2 dB.
+    //!
+    static constexpr uint8_t max_beep_volume = 0x1F;
+
+    //!  Describes the user callback.
+    //! \details NOTE: The user callback is called in IRQ context.
+    //! \param type Represents the type of event: half-transfer (HT) or transfer completed (TC).
+    //!
+    using user_callback = std::function< void (ecl::bus_event type) >;
+
+    //!  Codec low-level initialization routine.
+    //! \details This method performs low-level initialization of the
+    //!  codec and underlayer buses. This method must be called once
+    //!  before any operations with codec.
+    //! \retval Status of the operation.
+    //!
+    static ecl::err init();
+
+    //!  Codec power up sequence.
+    //! \details This method performs power up sequence of the
+    //!  codec which is recommended by CS43L22 Reference Manual (RM).
+    //!  To use this method, the init() method must be called previously.
+    //! \retval Status of the operation.
+    //!
+    static ecl::err power_up();
+
+    //!  Codec power down sequence.
+    //! \details This method performs power down sequence of the
+    //!  codec which is recommended by CS43L22 Reference Manual (RM).
+    //!  To use this method, the init() method must be called previously.
+    //! \retval Status of the operation.
+    //!
+    static ecl::err power_down();
+
+    //!  Send PCM buffer to the codec.
+    //! \details This method blocks until all data is transferred.
+    //!  The PCM data contains audio samples. Sample size is constant and equal to 16 bit.
+    //!  Sampling frequency can be changed with cs43l22< I2c, I2s, Rst_gpio >::set_sampling_frequency() method.
```

Comment:
I think doxygen is fine with simple `set_sampling_frequency()` without prepended class name